### ISSUE TITLE
Pre-trained weights

### ISSUE THREAD

**Issue by fdabrao** [AlexeyAB/darknet#4924]
Hello,
 
Do I realy need pre-trained weight file for training my custom object detections?
Why do I have to use one? What is the impact not using one?

Regards

**Comment by stephanecharette** [comment]
I never used the pre-trained weights.  I've always trained from scratch for my specific classes.  All it means is the weights files will be initialized with random values to start.  If you have a GPU and you are properly training with lots of images and enough iterations, I don't think it makes any difference.

**Comment by AlexeyAB** [comment]
Without pre-trained weights:
1. you should train 2x-4x more iterations
2. you should use 100 000 - 1 000 000 training images

**Comment by fdabrao** [comment]
> I never used the pre-trained weights. I've always trained from scratch for my specific classes. All it means is the weights files will be initialized with random values to start. If you have a GPU and you are properly training with lots of images and enough iterations, I don't think it makes any difference.

So, how do I start training without pre-treined weight?
`darknet detector train data/obj.data yolov3-tiny-obj.cfg yolov3-tiny.conv.15` ?

Regards

**Comment by Luxato** [comment]
@fdabrao 
Check this short video if you want to know what pre-training means and when you should use it.
https://www.youtube.com/watch?v=yofjFQddwHE

**Comment by Luxato** [comment]
> So, how do I start training without pre-treined weight?
> `darknet detector train data/obj.data yolov3-tiny-obj.cfg yolov3-tiny.conv.15` ?

If you don't want to use pre-trained weights then execute the command without them
 `darknet detector train data/obj.data yolov3-tiny-obj.cfg`

**Comment by mahaoyanghb** [comment]
@Luxato 
the original paper of YOLO make trainning into two parts: 
Part 1: choose a backbone classifer network, such as Alexnet,GoogleNet,Resnet
and make dataset for this backbone, and train it until get ideal performance.
Part 2: modify classifer backbone structure, mainly the last several layers, add regression layer for object detection, then labelling picture with bouding box , then train detection network with classifier pretrained weights(obtained from Part 1) 

you said just use the following command to train detection network from stretch.
darknet detector train data/obj.data yolov3-tiny-obj.cfg
can it work well ?
as far as I know object detection contain two part: feature extraction and location regression, 
with a feature extraction finished pretrained weight, location regression can be trained better, otherwise these two stage will do impact to each other, make trainning bad, can't get ideal result.

**Comment by Luxato** [comment]
Hey @mahaoyanghb,
I've never trained YOLO without pre-trained weights before, but as you could see what @stephanecharette wrote, it works fine.
Watch the video I posted above to see if pre-training makes sense for your case or not :).

**Comment by fdabrao** [comment]
Hello all,

At first, I trained to recognize 1 class for bathroom sink with 100 images and it took 2 days to have 0.06 avg loss, and I thought, is that maybe pre-trained weights has bad memory? Like learning other kind of images or learn other objects that I will not want to recognize? That´s because I asked if worth to use pre-trained. And besides, having yolo and yolo-tiny compare, the inference time is too different. So, my conclusion is that if I have "yolo-nano" it will be faster than all? But the result about train to recognize 1 class for bathroom sink was not good. In a video that with yolo-tiny I have 90-95% recognize with yolo-my-train was less than 1%?
So, for me it´s not clear what influences or how to use this as tool for the right choose to my project? Tell me if am I thinking in wrong way?

**Comment by jubaer-ad** [comment]
Can someone please check #4934?
